### DISCUSSION TITLE
nemo-preview improvement suggestion - allow file being previewed to update dynamically based on selection

### DISCUSSION THREAD

**Discussion by decuser** [linuxmint/discussions#20] [Ideas]
Hi,

Longtime mint user, first time caller :).

I recently made the switch to Linux Mint from MacOS across all of my PCs. In that process, I've been figuring out how different things work differently in Mint. One of those things is preview. On Mac, when you open Finder and select a file to preview, you press the space bar and up it pops. The same sort of thing happens on Mint. However, on the Mac, you can switch the item being previewed by selecting a different item while preview is running and it will be previewed. On Mint, you need to select the new item and re-press space.

My improvement suggestion is that nemo-preview be made aware of what the current selection in nemo is so that you can quickly change selections and see the results in the preview. 

My typical workflow is to open a window with items I'm interested in scanning through, selecting the first item in the list (a text file or image), press space, and then use the arrow keys to select the next and the next and the previous etc. All while preview merrily shows me the item selected without the necessity of having to re-hit space.

The equivalent workflow in Mint is to open a window with items I'm interested in scanning through, select the first item in the list (a text file or image), press space, and then select the nemo window again to get focus back, and then use the arrow keys to select the next, then press space again, and so on. It's a lot more work,

Hence the suggestion!

Thanks,

Will

**Comment by SebastJava** [comment]
Yeah, sure, but i would want some more basic concerns to be addressed first. I currently have an issue with thumbnails in Nemo but i still haven't found the time to systematically investigate and find the precise steps to reproduce and formally open a new issue in Nemo.

I don't remember what happened exactly but sometimes i have a window opened and there are SVG files with thumbnails in that window. I make some changes in Inkscape, and the thumbnail still displays the old version. So, i press Ctrl+R to reload (refresh) the window, but this doesn't work. The only thing that works is when i check this SVG file with Xviewer. There, in Xviewer, the image is up-to-date. And then, after going through Xviewer, when i press Ctrl+R in Nemo again, then Nemo finally wakes up.

(Sorry for this unwanted invasion, talking about my issues in your feature request, but i think it is related.)

**Reply by Jopp-gh** [reply]
Because the process "thumbnailer" is responsible to create and refresh the previews, Nemo doesn't. 

You can work with the thumbnailer, here an example
https://askubuntu.com/questions/1368910/how-to-create-custom-thumbnailers-for-nautilus-nemo-and-caja

**Comment by changenowde** [comment]
That would be super awesome to have in Nemo (also coming from a Mac - and I guess we are somewhat spoiled software-wise ...). I second this improvment suggestion!
